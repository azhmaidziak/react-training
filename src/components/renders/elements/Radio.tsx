import * as React from "react";
import { SearchBy } from "../../../models";

const Radio = ({
  input,
  onChange
}: {
  input: RadioInput;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}): React.ReactElement => (
  <React.Fragment>
    <input
      type="radio"
      id={input.id}
      name={input.name}
      value={input.value}
      onChange={onChange}
    />
    <label htmlFor={input.id}>{input.label}</label>
  </React.Fragment>
);

export default Radio;

export type RadioInput = {
  id: string;
  name: string;
  label: string;
  value: SearchBy;
};
