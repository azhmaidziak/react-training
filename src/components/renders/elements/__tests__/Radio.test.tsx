import * as React from "react";
import * as renderer from "react-test-renderer";
import {shallow} from "enzyme";
import Radio, {RadioInput} from "../Radio";

describe("Radio Element Test", () => {
    const input : RadioInput = {id:"id", name:"name", label:"label", value: 0};
    const onClick = jest.fn((e: React.ChangeEvent<HTMLInputElement>) => {
    });

    it('Test render correctly', () => {
        const testRenderer = renderer.create(<Radio input={input} onClick={onClick}/>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    it('Test click event', () => {
        const button = shallow(<Radio input={input} onClick={onClick}/>);
        button.find('input').simulate('click');
        expect(onClick.mock.calls.length).toEqual(1);
    });
});