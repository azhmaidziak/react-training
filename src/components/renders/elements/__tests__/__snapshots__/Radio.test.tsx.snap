// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Element Test Test render correctly 1`] = `
Array [
  <input
    id="id"
    name="name"
    onClick={[Function]}
    type="radio"
    value={0}
  />,
  <label
    htmlFor="id"
  >
    label
  </label>,
]
`;
